src/isDeepEqual.ts

```diff
@@ -121,6 +121,31 @@ function isDeepEqualImplementation<T, S>(data: S | T, other: S): data is S {
     return data.toString() === (other as unknown as RegExp).toString();
   }
 
+  if (data instanceof Map) {
+    if (data.size !== (other as unknown as Map<unknown, unknown>).size) {
+      return false;
+    }
+
+    const keys = Array.from(data.keys());
+
+    for (const key of keys) {
```

Comment:
**question:** is there a reason to not directly iterate over `data`?

```ts
for (const [key, value] of data) {
  // ...
}
```

Reply:
I wanted to do that initially but i got a ts error telling me that we should up the target to ES6 in the tsConfig in order to do that, should i do it ?

Reply:
no, we'll bump it in v2. But you don't need to use `Array.from` here...

Reply:
I can't iterate directly over data.keys() either because i get the same kind of ts error
What do you suggest, using  Object.keys or doing a for loop instead of using Array.from ?

Reply:
I pulled this PR, i can see that `Map.prototype.keys` returns an Iterator which needs to be spread in ES5 so your solution is ideal for ES5. Can you please add a TODO above the line saying that this should be fixed in v2. I'll follow up when I merge this PR into v2.